File: ch10p240subclassUIView/Empty Window/MyHorizLine.m


#import "MyHorizLine.h"

@implementation MyHorizLine

- (id)initWithCoder:(NSCoder *)decoder {
    self = [super initWithCoder:decoder];
    if (self) {
        self.backgroundColor = [UIColor clearColor];
    }
    return self;
}

- (void)drawRect:(CGRect)rect {
    CGContextRef c = UIGraphicsGetCurrentContext();
    CGContextMoveToPoint(c, 0, 0);
    CGContextAddLineToPoint(c, self.bounds.size.width, 0);
    CGContextStrokePath(c);
}

@end

File: iOS9bookExamples/bk1ch10p418subclassing/bk1ch10p418subclassing/MyHorizLine.swift


import UIKit

class MyHorizLine: UIView {

    required init?(coder aDecoder: NSCoder) {
        super.init(coder:aDecoder)
        self.backgroundColor = UIColor.clearColor()
    }
    override func drawRect(rect: CGRect) {
        let c = UIGraphicsGetCurrentContext()!
        CGContextMoveToPoint(c, 0, 0)
        CGContextAddLineToPoint(c, self.bounds.size.width, 0)
        CGContextStrokePath(c)
    }


}
